backend/logic/courses/courses.js

```diff
@@ -261,12 +261,102 @@ var updateCourse = async (course, newTitle, newTags) => {
   }
 };
 
+/**
+ * Assign a course to a curriculum
+ * @author Yufei Liu
+ * @param {string} curriculumType,{string} courseCode,{string} curriculum
+ * @returns true if successful
+ * @throws error if curriculumType, courseCode or curriculum dose not exist
+ *         course already assigned a curriculum
+ *
+ */
+var assignCourseToCurriculum = async (
+  curriculumType,
+  courseCode,
+  curriculum
+) => {
+  let conn = await mysql.getNewConnection();
+
+  const checkExistQuery = `SELECT *
+    FROM (
+        SELECT course_code AS course_code FROM curriculum_tech_comps
+        union all
+        SELECT course_code FROM curriculum_complementaries
+        union all
+        SELECT course_code FROM curriculum_core_classes
+    ) a
+    WHERE course_code = ?;`;
+
+  // Check if the course has already been assigned, verify if curriculum type, course and curriculum are existed in DB
+  let ifAssigned;
+  let checkCourse;
+  let checkCurriculum;
+
+  try {
+    ifAssigned = await conn.query(checkExistQuery, [courseCode]);
+    checkCourse = await conn.query(
+      "SELECT * FROM courses WHERE course_code = ?",
```

Comment:
```suggestion
      "SELECT COUNT(*) FROM courses WHERE course_code = ?",
```

Reply:
Since you are only using this to check if the course exists you dont need to select * which grabs all the columns in the table. You just need the count of rows that match your criteria